Include :datetime, :date, and :time with style options only (#1077)
unicode-org/message-format-wg: spec/functions/datetime.md

#### The `:datetime` function

> The function `:datetime` is used to format combined date/time values.

Reviewer: Perhaps clearer as:

```suggestion
The function `:datetime` is used to format both the date and time of a date/time value.
```

##### Options

The following _options_ are REQUIRED to be available on the function `:datetime`:

- `dateStyle`
  - `full`
  - `long`
  - `medium` (default)
  - `short`
- `timeStyle`
  - `full`
  - `long`
  - `medium`
  - `short` (default)
- _Date/time override options_

`dateStyle` and `timeStyle` are **_<dfn>style options</dfn>_**
which pertain to the overall styling or appearance of the formatted output.

If the _operand_ of the _expression_ is an implementation-defined date/time type,
it can include other option values.
Any _date/time override options_ of the operand are included in the resolved option values of the _expression_,
with _options_ on the _expression_ taking priority over any options of the _operand_.

> Any _operand_ options not matching the _Date/time override options_ are ignored.

Reviewer: formatting

```suggestion
Any _operand_ options not matching the _date/time override options_ are ignored.
```

If the _operand_ of the _expression_ is an implementation-defined date/time type,
it can include other option values.
Any _date/time override options_ of the operand are included in the resolved option values of the _expression_,
with _options_ on the _expression_ taking priority over any options of the _operand_.

> Any _operand_ options not matching the _Date/time override options_ are ignored.

Reviewer: ibid

Author: ```suggestion
Any _operand_ options not matching the _date/time override options_ are ignored.
```

If the _operand_ of the _expression_ is an implementation-defined date/time type,
it can include other option values.
Any _date/time override options_ of the operand are included in the resolved option values of the _expression_,
with _options_ on the _expression_ taking priority over any options of the _operand_.

> Any _operand_ options not matching the _Date/time override options_ are ignored.

Reviewer: ibid
